portal/views/extend_flask_user.py

```diff
@@ -40,3 +44,46 @@ def reset_password_view_function(token):
 
     next_url = url_for('user.reset_password', token=token)
     return challenge_identity(user_id=user_id, next_url=next_url)
+
+
+class LockoutLoginForm(LoginForm):
+    """adds lockout functionality to the login process"""
+
+    def validate(self):
+        """prevent locked out users from logging in
+
+        Before verifying the user's credentials
+        check to see if the user is locked out.
+        If the user is locked out display an error
+        message below the password field.
+        """
+        # Find user by email address (email field)
+        user_manager = current_app.user_manager
+        user, user_email = user_manager.find_user_by_email(self.email.data)
```

Comment:
I think it's convention to use `_` for "throwaway" variables, but an extra variable won't confuse there (and I otherwise never would've remembered that `find_user_by_email()` returns a tuple)